pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/nullvalue/NullValueIndexType.java

```diff
@@ -67,7 +70,7 @@ public NullValueVectorCreator createIndexCreator(IndexCreationContext context, I
 
   @Override
   public IndexConfig getDefaultConfig() {
-    return IndexConfig.DISABLED;
+    return IndexConfig.ENABLED;
```

Comment:
just to confirm:
1. this means null index type is always enabled, but it could contain no columns?
2. this has nothing to do with the schema definition JSONs `option.nullHandling.mode/default` value yes?

Reply:
> 1. this means null index type is always enabled, but it could contain no columns?

I don't think so. Columns contain indexes, not the other way arround

> 2. this has nothing to do with the schema definition JSONs option.nullHandling.mode/default value yes?

Nothing to do.

The reason to add this change is that before this change we always read the buffer, independently of the actual default config. Now we don't do that and therefore it is important to keep backward compatibility. Also, there are tons of tests that load segments without providing schema and table, which means that they use the default index config for all index types.